lxd/cluster/gateway.go

```diff
@@ -196,19 +197,27 @@ func (g *Gateway) HandlerFuncs(nodeRefreshTask func(*APIHeartbeat), trustedCerts
 			}
 
 			// Look for time skews
-			if heartbeatData.Time.Add(5 * time.Second).Before(time.Now().UTC()) {
-				if !g.timeSkew {
-					logger.Warn("Time skew detected between leader and local", log.Ctx{"leaderTime": heartbeatData.Time, "localTime": time.Now().UTC()})
-				}
-				g.timeSkew = true
-			} else if heartbeatData.Time.Add(-5 * time.Second).After(time.Now().UTC()) {
+			now := time.Now().UTC()
+
+			if heartbeatData.Time.Add(5*time.Second).Before(now) || heartbeatData.Time.Add(-5*time.Second).After(now) {
 				if !g.timeSkew {
-					logger.Warn("Time skew detected between leader and local", log.Ctx{"leaderTime": heartbeatData.Time, "localTime": time.Now().UTC()})
+					logger.Warn("Time skew detected between leader and local", log.Ctx{"leaderTime": heartbeatData.Time, "localTime": now})
+
+					err := g.Cluster.UpsertWarning("", "", -1, -1, db.WarningClusterTimeSkew, fmt.Sprintf("leaderTime: %s, localTime: %s", heartbeatData.Time, now))
```

Comment:
@monstermunchkin why is the nodeName argumet not populated here?

Reply:
Fixed